core/src/test/java/com/linecorp/armeria/client/PendingExceptionTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.EnumSource.Mode;
+
+import com.linecorp.armeria.common.ClosedSessionException;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.logging.RequestLogProperty;
+import com.linecorp.armeria.internal.client.PendingExceptionUtil;
+import com.linecorp.armeria.internal.testing.AnticipatedException;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+class PendingExceptionTest {
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.http(0);
+            sb.https(0);
+            sb.tlsSelfSigned();
+            sb.connectionDrainDurationMicros(0);
+            sb.service("/", (ctx, req) -> {
+                return HttpResponse.of(req.aggregate().thenApply(agg -> {
+                    ctx.log().ensureAvailable(RequestLogProperty.SESSION).channel().close();
+                    return HttpResponse.streaming();
+                }));
+            });
+        }
+    };
+
+    @EnumSource(value = SessionProtocol.class, mode = Mode.EXCLUDE, names = "PROXY")
+    @ParameterizedTest
+    void shouldPropagatePendingException(SessionProtocol protocol) {
+        final AnticipatedException pendingException = new AnticipatedException();
+        final BlockingWebClient client =
+                WebClient.builder(server.uri(protocol))
+                         .factory(ClientFactory.insecure())
+                         .decorator((delegate, ctx, req) -> {
+                             ctx.log().whenAvailable(RequestLogProperty.SESSION).thenAccept(log -> {
+                                 PendingExceptionUtil.setPendingException(log.channel(), pendingException);
+                             });
+                             return delegate.execute(ctx, req);
```

Comment:
Can you re-adjust the timeout so that each iteration doesn't take 10 seconds? (with a total of 1 minute for a single test)

Reply:
The long time was taken due to the graceful shutdown mechanism in the Netty HTTP/2 codec.
Configured the timeout to 1 second.  https://github.com/line/armeria/blob/e44512e3491a176e5ed3ee4ad8c945a3fd267dfb/core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java#L745-L746